app/src/main/java/com/sun/quickquotes/ui/home/HomeFragment.kt

```diff
@@ -1,27 +1,80 @@
 package com.sun.quickquotes.ui.home
 
+import android.annotation.SuppressLint
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.PagerSnapHelper
+import androidx.recyclerview.widget.RecyclerView
 import com.sun.quickquotes.R
 import com.sun.quickquotes.base.BaseFragment
+import com.sun.quickquotes.data.model.Quote
 import com.sun.quickquotes.databinding.FragmentHomeBinding
+import com.sun.quickquotes.ui.home.adapter.HomeQuoteAdapter
+import com.sun.quickquotes.utils.NotificationHelper
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class HomeFragment : BaseFragment<FragmentHomeBinding>() {
 
     override val layoutResource get() = R.layout.fragment_home
     override val viewModel: HomeViewModel by viewModel()
+    private var homeQuoteAdapter = HomeQuoteAdapter(::onFavoriteClick, ::onShareClick)
+    private val arg: HomeFragmentArgs by navArgs()
+    private var tagName: String? = null
 
     override fun initViews() {
-
+        viewBinding.recyclerViewHomeQuote.adapter = homeQuoteAdapter
     }
 
     override fun initData() {
         with(viewBinding) {
             lifecycleOwner = viewLifecycleOwner
             viewModel = this@HomeFragment.viewModel
+            val snapHelper = PagerSnapHelper()
+            snapHelper.attachToRecyclerView(recyclerViewHomeQuote)
+        }
+
+        val quote = arguments?.getParcelable<Quote>(NotificationHelper.KEY_OPEN_HOME_FRAGMENT)
+        tagName = arg.category
+        val name = tagName
+        when {
+            name != null -> viewModel.getDataByName(name)
+            quote != null -> viewModel.getDataFromNotify(quote)
+            else -> viewModel.getData()
         }
+
     }
 
+    @SuppressLint("NotifyDataSetChanged")
     override fun initEvents() {
+        viewBinding.recyclerViewHomeQuote.addOnScrollListener(object :
+            RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                viewModel.quotes.value?.let {
+                    val position =
+                        (viewBinding.recyclerViewHomeQuote.layoutManager as LinearLayoutManager)
+                            .findLastCompletelyVisibleItemPosition()
+                    if (position == it.size - 1) {
+                        if (tagName == null) {
+                            viewModel.getData()
+                        }
+                    }
+                }
+            }
+
+        })
+        viewModel.quotes.observe(this) {
```

Comment:
nên dùng this hay viewLiffcleOwner ?

Reply:
nên theo viewLifecycleOwner để tránh memory leak đúng không a, vì thằng view chết nhanh hơn thằng Fragment